pkg/controller/direct/compute/refs.go

```diff
@@ -154,9 +154,12 @@ func ResolveComputeAddress(ctx context.Context, reader client.Reader, src client
 
 	// targetField: address
 	// See compute servicemappings for details
-	address, _, err := unstructured.NestedString(computeAddress.Object, "spec", "address")
+	// Because `spec.address` field is optional, we can't guarantee it always
+	// exists in a successfully reconciled ComputeAddress CR, so we should use
+	// the `status.observedState.address` instead.
+	address, _, err := unstructured.NestedString(computeAddress.Object, "status", "observedState", "address")
```

Comment:
Clarification question: What if the user specifies an address? Will the `address` field be under `spec` then? Shall we check both `spec` and `status. observedState` for optional fields?

Reply:
`status.observedState.address` should be the same as the specified `spec.address` if ComputeAddress reconciles successfully, so I won't worry about the discrepancy.